onnxruntime/core/graph/contrib_ops/quantization_defs.cc

```diff
@@ -49,9 +50,9 @@ void ValidateTypeAndShapeForScaleAndZP(ONNX_NAMESPACE::InferenceContext& ctx, in
     }
   }
 
-  if (hasInputShape(ctx, index)) {
+  if ((expectedScalar != ScaleTensorType::Both) && hasInputShape(ctx, index)) {
```

Comment:
> if ((expectedScalar != ScaleTensorType::Both) && hasInputShape(ctx, index)) { [](http://example.com/codeflow?start=1&length=78)

we still need to check the shape for ScaleTensorType::Both like this way:
1. shape.dim_size() is in (0, 1)
2. if shape.dim_size() is 1. we can check the dim_value equal to expectedTensorSize.

Reply:
Done.